### PR TITLE
Add initial Dockerfiles, configs, and GitHub workflows for Cloudberry

### PR DESCRIPTION
**Pull request by edespino**
- Include Dockerfiles for Rocky Linux 8 and 9 based build and test environments
- Add test scripts, configuration files, and automation support for containerized builds
- Provide necessary scripts for testing and deploying across both supported environments
- Add GitHub workflow files for building and testing containersAdd initial Dockerfiles and configs for Cloudberry

### CODE REVIEW

File: images/docker/cbdb/build/rocky9/configs/init_system.sh [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+# - IP address
+# - CPU model name and number of cores
+# - Total memory available
+# This information is useful for users to understand the environment they
+# are working in.
+# --------------------------------------------------------------------
+source /etc/os-release
+
+cat <<-EOF
+Welcome to the Cloudberry Database Build Environment!
```
Comment (tuhaihe) [inline images/docker/cbdb/build/rocky9/configs/init_system.sh:150]: `Cloudberry Database` ~> `Apache Cloudberry` ?

**Review by tuhaihe — DISMISSED** [review]
Thank you, Ed! I just left some comments from a non-technical perspective for your information.

**Comment by tuhaihe** [comment]
The commit message can add a new line for the last one: 
`- Add GitHub workflow files for building and testing containers`
`- Add initial Dockerfiles and configs for Cloudberry`

File: images/docker/cbdb/test/rocky8/configs/init_system.sh [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+# - CPU model name and number of cores
+# - Total memory available
+# - Cloudberry version (if installed)
+# This information is useful for users to understand the environment they
+# are working in.
+# --------------------------------------------------------------------
+source /etc/os-release
+
+# Check if Cloudberry Database is installed and display its version
+if rpm -q cloudberry-db > /dev/null 2>&1; then
```
[the remark this answers is not in the document]
Comment (edespino) [reply images/docker/cbdb/test/rocky8/configs/init_system.sh]: I am changing the name to apache-cloudberry-db-incubating

**Review by tuhaihe — APPROVED** [review]

File: images/docker/cbdb/build/rocky8/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+# with other successful ASF projects.
+#
+# While incubation status is not necessarily a reflection of the
+# completeness or stability of the code, it does indicate that the
+# project has yet to be fully endorsed by the ASF.
+#
+# --------------------------------------------------------------------
+# Dockerfile for Apache Cloudberry Build Environment
+# --------------------------------------------------------------------
+# This Dockerfile sets up a Rocky Linux 9-based container for building
```
Comment (Zhangbaowen-Hashdata) [inline images/docker/cbdb/build/rocky8/Dockerfile:37]: Your contributions to the team have been invaluable. I just finished reading all the code. The process is very smooth and easy to understand. This is very helpful for me to understand the entire process. There is one place where I am wondering if it was a mistake. This Dockfile should be established as a Rocky Linux 8-based container for building.

Comment (edespino) [reply images/docker/cbdb/build/rocky8/Dockerfile:37]: @Zhangbaowen-Hashdata Thank you for the thorough review and kind feedback. I deliberately chose Rocky Linux 9 as our initial base platform. The plan is to expand support to additional platforms (including Rocky Linux 8, Ubuntu, and Debian) in upcoming iterations. For now, I wanted to focus on establishing solid automation processes for Apache Cloudberry (incubating) using a single platform as our foundation. Once we have that working smoothly, expanding to other distributions will be more straightforward.

**Pull request merged**